City-of-Helsinki/parking-permits-admin-ui
# Extend permit page

Should render price list on month changes

Refs: PV-782

## Description



## Context




[PV-782](https://helsinkisolutionoffice.atlassian.net/browse/PV-782)

## How Has This Been Tested?

Tested manually

## Manual Testing Instructions for Reviewers

Note: must use branch PV-782 of API.

If you navigate to a valid fixed period permit in the admin UI, the "Jatka tunnusta" button should appear along with the other action buttons at the bottom.

*Note* this button will not be shown if the permit cannot otherwise be extended: for example, there is another pending extension request on this permit.

Click on the button to go to the "Extend permit" page. Using the month counter control, add the number of desired months.

As you change the months, a table showing the dates and prices should appear, along with the total amount.

Clicking the "Cancel" button will return you back to the detail page with no changes.

Clicking the "Continue" button should open the modal to check for payment, clicking the "Confirm" button should add the months.

In the Django admin, the extension request should appear as "APPROVED" and the permit should be extended with the new end date.

## Screenshots

![Screenshot from 2024-02-21 12-05-40](https://github.com/City-of-Helsinki/parking-permits-admin-ui/assets/131681805/5627e391-8545-45bc-8354-4a929a055142)


[PV-782]: https://helsinkisolutionoffice.atlassian.net/browse/PV-782?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
